AllASPFuncs.py

```diff
@@ -1292,3 +1307,34 @@ def banner(text):
     length = len(text) + 8
     message = "\n" + "#" * length +'\n'+'#   ' + text + '   # \n'+ "#" * length
     Lager.debug(message)
+
+
+def prep_data_for_fit(images, err, sn_matrix, wgt_matrix):
+    '''
+    This function takes the data from the images and puts it into the form such
+    that we can analytically solve for the best fit using linear algebra.
+    '''
+    size = int(np.sqrt((images[0].size)))
+    tot_num = len(images)
+    det_num = len(sn_matrix)
+
+    # Flatten into 1D arrays
+    images = np.concatenate([arr.flatten() for arr in images])
+    err = np.concatenate([arr.flatten() for arr in err])
+
+    psf_zeros = np.zeros((np.size(images), tot_num))
+
+    for i in range(det_num):
+        psf_zeros[
+            (tot_num - det_num + i) * size * size:
+            (tot_num - det_num + i + 1) * size * size,
+            (tot_num - det_num) + i] = sn_matrix[i]
+    sn_matrix = psf_zeros
```

Comment:
You can replace these three lines with just

```
sn_matrix = np.vstack(psf_zeros)
```

psf_zeros is definitely an numpy array, you just created it above.  Also, for cases where you're not sure if something is an array or not, use `np.as_array` instead of `np.array`.  The former, `np.as_array` will only create a new object if necessary, where as `np.array` will always create a new array.  For small arrays, this doesn't matter, but you're doing linear-algebra where you should be starting to think about having large arrays and some of the performance implications.